### ISSUE TITLE
Questions on QoD user stories

### ISSUE THREAD

**Issue by emil-cheung** [camaraproject/rep_main#16]
1.	Is the following assumptions right?
-	Product Manager – A person within the enterprise that is consuming the API
-	Customer – A person that is using the application that the API consuming enterprise builds

2.	Is it not so sure that a developer would rather ask for a guaranteed low latency than a stable latency and a guaranteed bandwidth rather than a sable bandwidth? Using the term “stable” indicate a best effort scenario which is the starting point before asking for a specified QoD.

3.	Would a developer be aware of the term App-Flow? I could interpret it as a “single UE” to App data flow. I could also interpret it as a “All UE” to App data flow. And what about UE-UE communication?

4.	I can imagine two on-off cases. Either a time defined when the QoD is asked for (scenarios like a 15 min boost in gaming) or a turn on turn off ask in cases when a specific task needs to be performed (enhanced video quality during drone inspection work). Maybe it is better to have two user stories?

5.	I would expect the enterprise to want to know the actual delivered latency and bandwidth. Maybe continues measurements or at least a notification if the requested quality is not met to be able to take actions on the application side (warn the user of the application or switch to a fallback procedures to not risk failures)

6.	I feel we do not have the context of the stories around “decide later which package are ideally for billing”. Who is billing whom?

7.	The user stories on making it possible for the developer to compare if it was good to spend money on QoD (as compared to just using best effort) seems a bit strange. 
-	If the CSP can guarantee the quality as requested then the developer will get what he pays for and this should not be needed (if the CSP cannot guarantee the quality then this would be only a slightly better effort and why would you by such an offer in the first place)
-	The insights you would need is really the perceived end user quality (which includes other factors than pure communication)
-	The user story also indicates that to draw any conclusions you need to have measures on the same on perceived quality with both “standard best effort” as well as with enhanced QoD communication.

**Comment by shilpa-padgaonkar** [comment]
@emil-cheung : Please find some initial response to your questions below. For some questions/comments we will have a more detailed discussion in the WG call on 10th Dec

1. Product manager and api consumer roles mentioned in your presentation have been explained below:

- **Product Manager:** Product manager directs the production of APIs and fosters/cultivates a market of users. API product managers are also responsible for go-to-market strategy, API roadmap& lifecycle, budget ,customer service, making sure developers and end-users accessing the APIs get what they require.
- **Customer:** It is anyone consuming the API. It could result in several variations depending on the type API product eg. b2b, b2b2x etc.

2. These functionalities are based on relative priorities rather than guaranteed (blocked) resources. Stable is not equal to best effort and this can be experienced in cases of congestion.

3. Appflow will be explained as a part of the API documentation. This flow currently targets just single UE and the Application server.  The appflow can be quite simplistic with just source and destination IP address or can be quite finegrained where the user can specify not just source and destination ip address but also ports and protocols.  If there is any other usecase in the future, the current API can either be extended or a new API can be created.

4. IF I understand it right,  one is time based and other task based? so wud one have automatic timer for qos session deletion and other a manual one? The timer based feature could be easily added on the level of the Service Enabling Function, even if the NEF itself does not provide such feature and is on our roadmap. Currently duration is not intended to be mandatory for the consumer to specify. IF specified, timer based auto delete will take place and if not, it will either be deleted based on some max timeout or manual delete request.

5. If the question is directed towards which notifications (notifications as they would be available with the R16 AQP feature, i.e. things like: QOS_GUARANTEED / QOS_NOT_GUARANTEED) can an api consumer expect then we can surely discuss in the next call which notifications are relevant for the QoD API family.

6. Discussions about monetization models are out of scope as they would come under regulatory/antitrust issues.

7. We would need some more details here. Let's discuss in our next call.

8.  If "session" needs to change to something else can also be discussed in our next call

**Comment by NoelWirzius** [comment]
**Provide details what would differ in future regarding the 2 APIs which drove the decision to split the APIs in two**
  The goal of the service APIs is that they are so simple to use for developers with no background in telco industrie. In the end they should easily get what they want. Also the APIs can be very different in the integration into the network and for this we split this because they are two different customer needs one for latency and low bandwidth and the other for now high latency needs but high bandwidth needs.

**What happens when consumer needs combination of both?**
This is sth which depends on the implementation in the network. But in my mind this should not work. Letency API is just for low bandwidth and with the bandwidth you are not able to upgrade this. The Latency is from a priotization in the network always higher then the bandwidth API in my eyes. 

**Can the API be extended to provide the consumer with available profile options?**
This sounds great. I will recommend also to built an "info"-Api which is showing which APIs are currently avaible in the defined network cell or current location of the consumer. 
In my mind sth like this can be the result: 
- bandwidth api
- latency APi 
- Location monitoring 
- Customer  informtation
- Slicing 
- ... 
- ... 
 
so: API-Detailinfo Call and General API avaibility API

**Comment by jordonezlucena** [comment]
We should follow user story template discussed in Commonalities WG for every selected use case. Otherwise, some of the details/implications of the user stories are not fully clear.

**Comment by shilpa-padgaonkar** [comment]
Yes fully agree. We will submit a revised version of the user stories once the user story commonalities template is finalized.
